### DISCUSSION TITLE
What is the main difference between sensor/⁠framelinvel and sensor/⁠velocimeter?

### DISCUSSION THREAD

**Discussion by dogged1021** [google-deepmind/mujoco#2296] [Asking for Help] [answered]
### Intro

Hi! I am an enthusiast interested in robotics and have just started to learn mujoco.

### My setup

Mujoco 3.2.6
API C
ubuntu 20.04

### My question

What is the main difference between sensor/⁠framelinvel and sensor/⁠velocimeter in a XML model? Can I simply regard sensor/⁠velocimeter as a sensor/⁠framelinvel with the frame-of-reference itself and the site ref-type? And I have the same confusion about sensor/⁠frameangvel and sensor/⁠gyro.
I have searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions) but still not sure. Thanks for your time!

### Confirmations

- [x] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment, marked as answer]
Not exactly. If you define a framevel of a thing w.r.t itself you will get zero by definition; nothing moves in its own frame. the velocimeter and gyro are w.r.t to a frame that is oriented in the same way but is **not moving**. Read engine_sensor.c for full details.

**Reply by dogged1021** [reply]
Thanks a lot! And I would like to read  engine_sensor.c later.
Can I understand that sensor/⁠framelinvel and sensor/⁠velocimeter return the instantaneous velocity and angular velocity w.r.t the current local frame (namely itself)?

**Reply by yuvaltassa** [reply]
Yes, w.r.t the current local frame, but one that is **not moving**.

**Reply by dogged1021** [reply]
OK, thanks again!
